File: PHP/leetcode/sort-integers-by-the-number-of-1-bits/solution.php
class Solution {
    function sortByBits($arr) {
        usort($arr, function($a, $b) {
            $pa = substr_count(decbin($a), '1');
            $pb = substr_count(decbin($b), '1');
            if ($pa != $pb) return $pa - $pb;
            return $a - $b;
        });
        return $arr;
    }
}

File: CSharp/leetcode/sort-integers-by-the-number-of-1-bits/solution.cs
public class Solution {
    public int[] SortByBits(int[] arr) {
        return arr.OrderBy(x => CountBits(x)).ThenBy(x => x).ToArray();
    }
    private int CountBits(int x) {
        int c = 0;
        while (x > 0) { c += x & 1; x >>= 1; }
        return c;
    }
}
